### PR TITLE
keymaps: add kinesis/keymaps/jwon

### PR DESCRIPTION
**Pull request by jwon**
## Description

Created my own personal keymap for the Kinesis Advantage 2 using the Stapelberg mod.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kinesis/keymaps/jwon/config.h [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (drashna) [inline keyboards/kinesis/keymaps/jwon/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kinesis/keymaps/jwon/keymap.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kinesis/keymaps/jwon/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kinesis/keymaps/jwon/keymap.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        if (get_mods() & MOD_MASK_SHIFT) {
+            writePinLow(LED_CAPS_LOCK_PIN);
+        }
+    } else {
+        writePinHigh(LED_CAPS_LOCK_PIN);
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/kinesis/keymaps/jwon/keymap.c:166]: You could do this, too:
```suggestion
void matrix_scan_user(void) {
    writePin(LED_CAPS_LOCK_PIN, !(get_mods() & MOD_MASK_SHIFT));
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by jwon** [comment]
Thank you for the feedback @drashna !

I made the changes you suggested. Can you please take another look?

**Review by drashna — APPROVED** [review]

File: keyboards/kinesis/keymaps/jwon/config.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2021 James Won <me@jwon.me>
```
Comment (fauxpark) [inline keyboards/kinesis/keymaps/jwon/config.h:1]: This file can actually be removed entirely.

File: keyboards/kinesis/keymaps/jwon/rules.mk [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
```
Comment (fauxpark) [inline keyboards/kinesis/keymaps/jwon/rules.mk:18]: Most of this can be removed, you only need the differences from the keymap/revision level rules.mk.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by jwon** [comment]
@fauxpark I made the changes you requested. Can you please take another look? Thank you very much!

File: keyboards/kinesis/keymaps/jwon/rules.mk [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in 
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (fauxpark) [inline keyboards/kinesis/keymaps/jwon/rules.mk:7]:
```suggestion
BOOTMAGIC_ENABLE = no
MOUSEKEY_ENABLE = no
NKRO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**